main.cpp

```diff
@@ -47,58 +52,55 @@ int main(int argc, char *argv[]) {
 
     int mode = atoi(argv[2]);
     std::string JASMINEGRAPH_HOME = Utils::getJasmineGraphHome();
-    std::string profile = argv[1];  // This can be either "docker" or "native"
+    jasminegraph_profile = strcmp(argv[1], "docker") == 0 ? PROFILE_DOCKER : PROFILE_K8S;
     std::string enableNmon = "false";
 
     main_logger.log("Using JASMINE_GRAPH_HOME", "info");
     std::cout << JASMINEGRAPH_HOME << std::endl;
 
+    StatisticCollector::init();
+    thread schedulerThread(SchedulerService::startScheduler);
+
     if (mode == Conts::JASMINEGRAPH_RUNTIME_PROFILE_MASTER) {
         std::string masterIp = argv[3];
         int numberOfWorkers = atoi(argv[4]);
         std::string workerIps = argv[5];
         enableNmon = argv[6];
         server = JasmineGraphServer::getInstance();
-        thread schedulerThread(SchedulerService::startScheduler);
 
-        if (profile == Conts::PROFILE_K8S) {
-            K8sInterface* interface = new K8sInterface();
-            masterIp = interface->getMasterIp();
+        if (jasminegraph_profile == PROFILE_K8S) {
+            std::unique_ptr<K8sInterface> k8sInterface(new K8sInterface());
+            masterIp = k8sInterface->getMasterIp();
             if (masterIp.empty()) {
-                masterIp = interface->createJasmineGraphMasterService()->spec->cluster_ip;
+                masterIp = k8sInterface->createJasmineGraphMasterService()->spec->cluster_ip;
             }
         }
-        server->run(profile, masterIp, numberOfWorkers, workerIps, enableNmon);
-
-        while (server->isRunning()) {
-            usleep(microseconds);
-        }
+        server->run(masterIp, numberOfWorkers, workerIps, enableNmon);
 
         schedulerThread.join();
         delete server;
     } else if (mode == Conts::JASMINEGRAPH_RUNTIME_PROFILE_WORKER) {
         main_logger.log(to_string(argc), "info");
 
-        if (argc < 5) {
-            main_logger.log("Need Four arguments. Use 2 <hostName> <serverPort> <serverDataPort> to start as worker",
-                            "info");
+        if (argc < 8) {
```

Comment:
If we define an enum listing the arguments we can replace 8 with that particular enum size.

Reply:
Ok